common/src/main/java/com/spotify/spydra/util/GcpConfiguration.java

```diff
@@ -0,0 +1,137 @@
+package com.spotify.spydra.util;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.ServiceOptions;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.PathNotFoundException;
+
+public abstract class GcpConfiguration {
+
+  public static final GcpConfiguration create() {
+      return useApplicationDefaultCredentials()
```

Comment:
I think this can be simplified, but I'm not certain. When you do `GoogleCredentials.getApplicationDefault();` that will go through: 
   *   Credentials file pointed to by the {@code GOOGLE_APPLICATION_CREDENTIALS} environment variable</li>
   *   Credentials provided by the Google Cloud SDK
   *   Google App Engine built-in credentials
   *   Google Cloud Shell built-in credentials
   *   Google Compute Engine built-in credentials
And if that throws an IOException you know that the credential could not be created in that environment, if so try to create from a provided service account key. Or am I missing something? 

Granted you still need to get the project from that file, which means you still need to parse it if it's set.

Reply:
That was what I hoped for when starting this patch. Unfortunately the gcs-connector library does not search for credentials this way[1]. It will use a metadata server if found, but otherwise require a key file. 

It may be possible to remove the usage of that lib and simplify. I can give this another look. 

1: https://github.com/GoogleCloudPlatform/bigdata-interop/blob/master/util/src/main/java/com/google/cloud/hadoop/util/CredentialConfiguration.java#L64

Reply:
Ah, ok, then it's fine. Let's not add more lead time to this PR. It still follows the patterns used earlier in the code.